patches/0003-Implement-crypto-internal-backend.patch

```diff
@@ -356,6 +418,31 @@ index e913f9885234fd..ef331980a816a3 100644
  	appendSetting("-buildmode", buildmode)
  	appendSetting("-compiler", cfg.BuildContext.Compiler)
  	if gccgoflags := BuildGccgoflags.String(); gccgoflags != "" && cfg.BuildContext.Compiler == "gccgo" {
+diff --git a/src/cmd/go/internal/tool/tool.go b/src/cmd/go/internal/tool/tool.go
+index 16e1a4f47f49c0..38688015d5a704 100644
+--- a/src/cmd/go/internal/tool/tool.go
++++ b/src/cmd/go/internal/tool/tool.go
+@@ -281,6 +281,10 @@ func buildAndRunBuiltinTool(ctx context.Context, toolName, tool string, args []s
+ 	// Override GOOS and GOARCH for the build to build the tool using
+ 	// the same GOOS and GOARCH as this go command.
+ 	cfg.ForceHost()
++	// The tool might build some additional packages, set the GOEXPERIMENT
++	// environment variable to the one we just computed in cfg.ForceHost.
++	originalGOEXPERIMENT := os.Getenv("GOEXPERIMENT")
++	os.Setenv("GOEXPERIMENT", cfg.RawGOEXPERIMENT)
+ 
+ 	// Ignore go.mod and go.work: we don't need them, and we want to be able
+ 	// to run the tool even if there's an issue with the module or workspace the
+@@ -288,6 +292,9 @@ func buildAndRunBuiltinTool(ctx context.Context, toolName, tool string, args []s
+ 	modload.RootMode = modload.NoRoot
+ 
+ 	runFunc := func(b *work.Builder, ctx context.Context, a *work.Action) error {
++		// Now that the tool is built, revert the GOEXPERIMENT
++		// environment variable to the original value.
++		os.Setenv("GOEXPERIMENT", originalGOEXPERIMENT)
```

Comment:
Does this func ever get called in parallel? I assume not, but if so, seems like it would be easy to get "stuck" with a GOEXPERIMENT value and end up with flakiness.